src/Interfaces/LazySet.jl

```diff
@@ -682,3 +683,44 @@ end
 
 isoperationtype(::LazySet) = error("`isoperationtype` cannot be applied to " *
                                    "a set instance; use `isoperation` instead")
+
+"""
+    isequivalent(X::LazySet, Y::LazySet)
+
+Return whether two LazySets are equal in the mathematical sense, i.e. equivalent.
+
+### Input
+
+- `X` -- any `LazySet`
+- `Y` -- another `LazySet`
+
+### Output
+
+`true` iff `X` is equivalent to `Y`.
+
+## Algorithm
+
+First, the check `X == Y` is performed which returns `true` if and only if the given sets are of the same type,
+and have the same values (modulo floating-point tolerance). Otherwise, the double inclusion check `X ⊆ Y && Y ⊆ X` is
+computed.
+
+### Examples
+
+```jldoctest
+julia> X = BallInf([0.1, 0.2], 0.3);
+
+julia> Y = convert(HPolytope, X);
+
+julia> X == Y
+false
+
+julia> isequivalent(X, Y)
+true
+```
+"""
+function isequivalent(X::LazySet, Y::LazySet)
+    if X == Y
```

Comment:
Since the inclusion checks are approximate, it could be more consistent to use `X ≈ Y` here. At least this would match the documentation. (So at least the documentation or the code have to be changed here.)
However, `≈` is not controlled by our internal precision variables, so it might also be confusing. Thoughts?

Reply:
I would suggest adding `≈`, then it fits the documentation.

If you want to make it less confusing, we could also just remove this preceding equality check.

Reply:
> I would suggest adding ≈, then it fits the documentation.

I think this is as good idea :thumbsup:

> If you want to make it less confusing, we could also just remove this preceding equality check.

We shouldn't remove this initial comparison: if the sets are of the same type, it is going to be more efficient than the double inclusion check.

> However, ≈ is not controlled by our internal precision variables

Good point, that should be fixed by using `_isapprox` in line 437.

Reply:
> Good point, that should be fixed by using _isapprox in line 437.

This is now https://github.com/JuliaReach/LazySets.jl/pull/1824